test/HttpApiHandlerTests.cs

```diff
@@ -1,202 +1,272 @@
+        }
+
+        [Fact]
+        public async Task WaitForCompletionOrCreateCheckStatusResponseAsync_Returns_Defaults_When_Runtime_Status_is_Canceled()
+        {
+            await this.CheckRuntimeStatus(TestConstants.InstanceIdCanceled, OrchestrationRuntimeStatus.Canceled);
+        }
+
+        private async Task CheckRuntimeStatus(string instanceId, OrchestrationRuntimeStatus runtimeStatus)
+        {
+            var httpApiHandler = new HttpApiHandler(new DurableTaskExtensionMock() { NotificationUrl = new Uri(TestConstants.NotificationUrl) }, null);
+            var httpResponseMessage = await httpApiHandler.WaitForCompletionOrCreateCheckStatusResponseAsync(
+                new HttpRequestMessage
+                {
+                    RequestUri = new Uri(TestConstants.RequestUri),
+                },
+                instanceId,
+                new OrchestrationClientAttribute
+                {
+                    TaskHub = TestConstants.TaskHub,
+                    ConnectionName = TestConstants.ConnectionName,
+                },
+                TimeSpan.FromSeconds(30),
+                TimeSpan.FromSeconds(8));
+            Assert.Equal(httpResponseMessage.StatusCode, HttpStatusCode.OK);
+            var content = await httpResponseMessage.Content.ReadAsStringAsync();
+            var response = JsonConvert.DeserializeObject<JObject>(content);
+            Assert.Equal(response["runtimeStatus"], runtimeStatus.ToString());
+        }
+
+        [Fact]
+        public async Task TerminateInstance()
+        {
+            string testInstanceId = Guid.NewGuid().ToString("N");
+            string testReason = "TerminationReason" + Guid.NewGuid();
+
+            string actualInstanceId = null;
+            string actualReason = null;
+
+            var clientMock = new Mock<DurableOrchestrationClientBase>();
+            clientMock
+                .Setup(x => x.TerminateAsync(It.IsAny<string>(), It.IsAny<string>()))
+                .Returns(Task.CompletedTask)
+                .Callback((string instanceId, string reason) =>
+                {
+                    actualInstanceId = instanceId;
+                    actualReason = reason;
+                });
+
+            clientMock
+                .Setup(x => x.GetStatusAsync(It.IsAny<string>(), It.IsAny<bool>(), It.IsAny<bool>()))
+                .Returns(Task.FromResult(
+                    new DurableOrchestrationStatus
+                    {
+                        InstanceId = testInstanceId,
+                        RuntimeStatus = OrchestrationRuntimeStatus.Running,
+                    }));
+
+            var terminateRequestUriBuilder = new UriBuilder(TestConstants.NotificationUrl);
+            terminateRequestUriBuilder.Path += $"/Instances/{testInstanceId}/terminate";
+            terminateRequestUriBuilder.Query = $"?reason={testReason}&{terminateRequestUriBuilder.Query.TrimStart('?')}";

```

Comment:
UriBuilder.Query automatically prepends a ? to whatever is assigned to it. The current code will result in "{path}/??reason={etc}".

Reply:
Hmm...when i looked at it in the debugger I thought it was resolving correctly. I'll look at it again. I'll admit I didn't really understand the behavior of this API.